components/renku_data_services/base_models/core.py

```diff
@@ -21,6 +21,44 @@ async def authenticate(self, access_token: str, request: Request) -> "APIUser":
         ...
 
 
+# @final
+# class Id:
+#     """Id is a descriptor that wraps a string id in class attributes."""
+
+#     def __init__(self, id: str) -> None:
+#         if id is None:
+#             raise ValueError("Id cannot be None")
+#         self._id = id
+
+#     def __get__(self, obj, objtype=None) -> str:
+#         return self._id
+
+#     def __set__(self, obj, value: str):
+#         if value is None:
+#             raise ValueError("Id cannot be None")
+#         self._id = value
+
+
+# @final
+# class NoId:
+#     """A descriptor for wrapping non-existing ids in class attributes.
+
+#     Fails on access.
+#     """
+
+#     def __get__(self, obj, objtype=None):
+#         raise NotImplementedError("Cannot get an id of NoId")
+
+#     def __set__(self, obj, value):
+#         raise NotImplementedError("Cant set NoId, use dedicated 'set_id' method instead")
```

Comment:
I think all of this can be removed, right @Panaetius?

Reply:
ah yes, that was an earlier attempt using descriptors, but mypy doesn't understand them properly and it's simple with typevar anyways